hack/scripts/sync-prow.sh

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+check_deps() {
+  if ! which git-filter-repo > /dev/null 2>&1; then
+    echo "please install the git-filter-repo tool per https://github.com/newren/git-filter-repo"
+    exit 1
+  fi
+}
+
+check_deps
+
+old_prow="$PWD/test-infra"
+new_prow="$PWD/prow"
+
+# Clone the old prow repo.
+if [[ ! -d "$old_prow" ]]; then
+    git clone https://github.com/kubernetes/test-infra.git
+fi
+
+# Strip down the master branch of $old_prow to only have commits that touch the the /prow directory.
+pushd "$old_prow"
+git-filter-repo --path prow
+popd
```

Comment:
It's not clear why pushd/popd are needed, and it sort of confused me into thinking $new_prow is a relative path to the original directory.

Can use plan "cd" above and omit popd since both $old_prow and $new_prow are absolute by virtue of $PWD.

Reply:
Done.

Reply:
> It's not clear why pushd/popd are needed, and it sort of confused me into thinking $new_prow is a relative path to the original directory.

It's because when I originally wrote the script I didn't use `$old_prow` or other variables. Thanks for the suggestion.